apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
Heh,  just two preliminary things after quick look:

* We'll likely get rid of ```store``` event once we migrate to Drafts service that has ```save``` and ```delete``` promise-based methods and ```deleted``` and ```saved``` events - for other parties who didn't call those methods directly, but want to know when such operation happen.  I suppose that both ```delete``` and ```save``` should call ```store``` internally (unless in batching mode, but it's another story) and events should be fired only once it's actually stored. What do you think? 

* I don't really like promise-based ```EventDispatcher.once``` :) IMO, it feels like not-in-the-right place, mixing even- and promise- based flows...

But I'll take a look deeper on Monday, maybe I'm plain wrong :)